### ISSUE TITLE
Enforce imported partial file extension

### ISSUE THREAD

**Issue by amstiel** [kristerkari/stylelint-scss#349]
Hi! Is there a way to enforce partial file extension specifying?
For example:
`@import 'style.scss';` - valid
`@import 'style';` - not valid

I tried to use `"scss/at-import-partial-extension-whitelist"` rule, but it works only when extension is specified. My goal is to forbid importing `.sass` files. Whitelist rule does not work for `.sass` file which imported like a `@import 'sassfile';`.

**Comment by nex3** [comment]
Writing explicit extensions is not something I'd consider good style. Why enforce this at the `@import` site rather than enforcing that no indented-syntax files exist?

**Comment by amstiel** [comment]
So why does rule `"scss/at-import-partial-extension-whitelist"` exist if it does not work when I do not specify an extension? And why there is no rule that would force the developer to specify an extension?
By the way `sass-lint` has rule `filename-extension` for this.

**Comment by kristerkari** [comment]
> So why does rule `"scss/at-import-partial-extension-whitelist"` exist if it does not work when I do not specify an extension?

That's most likely something that needs to be fixed. The other `whitelist` rules in stylelint allow the user to pass `[]` as the option.

**Comment by amstiel** [comment]
@kristerkari Perhaps you did not understand me correctly, I mean explicit extensions of imported files, not value of rule. And if value of blacklist rule is ['sass'], you can't `@import 'indented_sass_file.sass`, but you still can `@import 'indented_sass_file`. So, why this rule 'blacklist' exist, if you can bypass this rule by simply removing the extension of imported file?

Or, maybe, `@import "path/fff";` pattern should be considered warning of the `at-import-partial-extension-whitelist` rule? Because extension of this "fff" file is ambiguous and we do not know whether it is white listed or not

**Comment by kristerkari** [comment]
Okay, I think that there are two things getting mixed up here. 

The extension blacklist/whitelist rules are only meant to restrict the extensions that you are allowed to use in your files.

To fix the issue that you are facing, there should be a new rule `at-import-partial-extension` with options `always`/`never`.

That way you can do this:

```json
"scss/at-import-partial-extension": "always",
"scss/at-import-partial-extension-whitelist": ["scss"]
```

What do you think @amstiel?

**Comment by kristerkari** [comment]
Some examples

```json
"scss/at-import-partial-extension": "always",
"scss/at-import-partial-extension-whitelist": ["scss"]
```

```scss
@import "./style"; // not valid
@import "./style.sass"; // not valid
@import "./style.scss"; // valid
```

---

```json
"scss/at-import-partial-extension": "never"
```

```scss
@import "./style"; // valid
@import "./style.scss"; // not valid
```

**Comment by amstiel** [comment]
@kristerkari Awesome 😄. That's what I'm talking about
